packages/opentelemetry-instrumentation-ws/src/index.ts

```diff
@@ -419,8 +422,11 @@ export class WSInstrumentation extends InstrumentationBase<WS> {
               parentSpan?.setAttributes({
                 [SemanticAttributes.HTTP_STATUS_CODE]: 101,
               });
-              parentSpan?.end();
-              self._requestSpans.delete(request);
+              if (parentSpan) {
+                const actualEndTime = new Date();
+                const originalEnd = parentSpan.end;
+                parentSpan.end = (_endtime?: TimeInput) => originalEnd(actualEndTime); // Set the end time to the current time but don't end the span yet
```

Comment:
Did you test locally to see that the parent span is actually ended by something or other? What does actually end it for a WS request in fastify?